src/ray/core_worker/store_provider/plasma_store_provider.cc

```diff
@@ -126,9 +127,22 @@ Status CoreWorkerPlasmaStoreProvider::FetchAndGetFromPlasmaStore(
   return Status::OK();
 }
 
+Status UnblockIfNeeded(const std::shared_ptr<RayletClient> &client,
+                       const WorkerContext &ctx) {
+  if (ctx.CurrentTaskIsDirectCall()) {
+    if (ctx.ShouldReleaseResourcesOnBlockingCalls()) {
+      return client->NotifyDirectCallTaskUnblocked();
+    } else {
+      return Status::OK();  // no-op
```

Comment:
Can you add a comment explaining under what cases we reach this line?

Reply:
Fixing.